flask_admin/templates/bootstrap4/admin/layout.html

```diff
@@ -45,7 +45,7 @@
               {% else %}
               <li{% if class_name %} class="{{class_name}}"{% endif %}>
               {%- endif %}
-              <a class="nav-link" href="{{ child.get_url() }}"{% if child.target %}
+              <a class="nav-link text-reset" href="{{ child.get_url() }}"{% if child.target %}
```

Comment:
Using this combination of classes, the element will still look incorrect. Try this:
```html
<a class="dropdown-item" href="{{ child.get_url() }}"{% if child.target %}
```

Reply:
Do you mind sharing a screenshot or the name of the swatch you are using?

Reply:
I'm not using swatch to display items faithfully, pure bootstrap. This statement is based on documentation of bootstrap. Considering the need to display activated items with class `active`. The correct markup in this block will look like this:
```jinja2
          <ul class="dropdown-menu">
          {%- for child in children -%}
            {%- if child.is_category() -%}
              {{ menu(menu_root=[child]) }}
            {% else %}
              {% set class_name = child.get_class_name() %}
              <li{% if class_name %} class="{{class_name}}"{% endif %}>
              {%- if child.is_active(admin_view) %}
              <a class="dropdown-item active" href="{{ child.get_url() }}"{% if child.target %}
               target="{{ child.target }}"{% endif %}>
              {{ menu_icon(child) }}{{ child.name }}</a>
              {% else %}
              <a class="dropdown-item" href="{{ child.get_url() }}"{% if child.target %}
               target="{{ child.target }}"{% endif %}>
              {{ menu_icon(child) }}{{ child.name }}</a>
              {%- endif %}
              </li>
            {%- endif %}
          {%- endfor %}
          </ul>
```
And here is screenshot for my variant:
![изображение](https://user-images.githubusercontent.com/37049574/89709214-99f47000-d986-11ea-9393-e2d512252f2f.png)

Background of main navbar is dark because I fixed the classes of navbar considering to bootstrap documentation. Here is my navbar classes markup:
```jinja
        <nav class="navbar navbar-expand-lg navbar-dark bg-dark mb-2" role="navigation">
```

And thanks for the support in integration of bs4

Reply:
Oh. I see. Let me integrate these changes in the PR.

Reply:
So we are going for default of a dark background?

Reply:
I think dark navbar looks better.

Reply:
Done.